addons/website/models/assets.py

```diff
@@ -60,33 +60,21 @@ def _get_custom_attachment(self, custom_url, op='='):
         res = super(Assets, self)._get_custom_attachment(custom_url, op=op)
         return res.with_context(website_id=website.id).filtered(lambda x: not x.website_id or x.website_id == website)
 
-    def _get_custom_view(self, custom_url, op='='):
+    def _get_custom_asset(self, custom_url):
         """
-        See web_editor.Assets._get_custom_view
+        See web_editor.Assets._get_custom_asset
         Extend to only return the views related to the current website.
         """
         website = self.env['website'].get_current_website()
-        res = super(Assets, self)._get_custom_view(custom_url, op=op)
-        return res.with_context(website_id=website.id).filter_duplicate()
+        res = super(Assets, self)._get_custom_asset(custom_url)
+        return res.with_context(website_id=website.id).filtered(lambda x: not x.website_id or x.website_id == website)
```

Comment:
Filter duplicate return the more specific
Here you will have duplicate generic & specific